R/get_gbif.R

```diff
@@ -461,16 +462,35 @@ get_gbif <- function(sp_name = NULL,
 
 		## Try the download first: may be request overload problems
 		go.tile <- geo.ref[x]
-		gbif.search <- try(
-			rgbif::occ_data(taxonKey = sp.key,
-											limit = occ_samp,
-											hasCoordinate = !no_xy,
-											hasGeospatialIssue = FALSE,
-											geometry = go.tile),
-			silent=TRUE)
+		gbif.search <- if (should_use_occ_download) {
+			req_id = rgbif::occ_download(
+				rgbif::pred("taxonKey", sp.key),
+				rgbif::pred("hasCoordinate", !no_xy),
+				rgbif::pred("hasGeospatialIssue", FALSE),
+				rgbif::pred_within(go.tile),
+				format = "SIMPLE_CSV",
+				curlopts=list(http_version=2)
+			)
+			cat(">>> Download request ID:", req_id, "\n")
+			rgbif::occ_download_wait(req_id, status_ping = 5, curlopts = list(http_version=2), quiet = FALSE)
+			download <- rgbif::occ_download_get(req_id)
+			try(rgbif::occ_download_import(download), silent = FALSE)
+		} else {
+    	cat(">>>> #", x, " (", round(x * 100/length(geo.ref), 2), "%): ", sep="")
+			try(
+				rgbif::occ_data(
+					taxonKey = sp.key,
+					limit = occ_samp,
+					hasCoordinate = !no_xy,
+					hasGeospatialIssue = FALSE,
+					geometry = go.tile
+				),
+				silent=TRUE
+			)
+		}
 
 		# If problems, just try to rerun with while with n attempts, otherwise return NULL
-		if (class(gbif.search) %in% "try-error") {
+		if (!should_use_occ_download && class(gbif.search) %in% "try-error") {
```

Comment:
Errors look different here for occ_download. I just skipped this block entirely. Not sure if there's a better way to check for errors but I don't really expect any here